cms/middleware.py

```diff
@@ -37,11 +37,13 @@ def process_request(self, request):
         if not any(pattern.match(request.path_info[1:]) for pattern in exclude_urls):
             # See if preview mode is requested.
             try:
-                preview_mode = bool(int(request.GET.get('preview', 0)))
+                path = f'{request.path_info[1:] if request.path_info[1:] else request.path_info}'
+                token_preview_valid = path_token_generator.check_token(request.GET.get('preview', 0), path)
+                user_preview = request.GET.get('preview', 0) and request.user.is_authenticated() and request.user.is_staff
```

Comment:
Change `is_authenticated()` to `is_authenticated` while you are here - the callable version was deprecated in 1.11 and removed in 2.x.